docs/source/notes/faq.rst

```diff
@@ -0,0 +1,72 @@
+Frequently Asked Questions
+==========================
+
+My model reports "cuda runtime error(2): out of memory"
+-------------------------------------------------------
+
+As the error message suggests, you have run out of memory on your
+GPU.  Since we often deal with large amounts of data in PyTorch,
+small mistakes can rapidly cause your program to use up all of your
+GPU; fortunately, the fixes in these cases are often simple.
+Here are a few common things to check:
+
+**Don't accumulate history across your training loop.**
+By default, computations involving variables that require gradients
+will keep history.  This means that you should avoid using such
+variables in computations which will live beyond your training loops,
+e.g., when tracking statistics; you should instead detach the variable
+or access its data correctly.
+
+Sometimes, it can be non-obvious when differentiable variables can
+occur.  Consider the following training loop (abridged from `source
+<https://discuss.pytorch.org/t/high-memory-usage-while-training/162>`_):
+
+.. code::
+
+    total_loss = 0
+    for i in range(10000):
+        optimizer.zero_grad()
+        output = model(input)
+        loss = criterion(output)
+        loss.backward()
+        optimizer.step()
+        total_loss += loss[0]
+
+Here, ``total_loss`` is holding on to history across all of the training
```

Comment:
Not necessarily. Shouldn't `loss.backward()` free the graph? IIUC, it should only hold the graph
```
loss 1 \
loss 2 - total_loss
...
loss k /
```

Reply:
That's a good point. I wonder why they were running out of memory in the discussion post. We can run the experiment.

Reply:
Very likely due to large Linear (and maybe long sequence). It's a Siamese network so that single layer is taking 2*2*1 = 4G memory per iteration.

Reply:
Even in that's the case, it's still a bad idea to keep the graphs alive.

Reply:
@apaszke  That's definitely true. I was just saying that it's not holding on to history of all training iterations.

Reply:
Sure, what you said makes a lot of sense! I wasn't trying to deprecate that in any way, sorry if it ended up looking like this.

Reply:
No no I'm sorry. I wasn't getting that impression at all. Thanks for the clarification still! :)